build/jsdoc-typeof-plugin.js

```diff
@@ -3,6 +3,16 @@
  */
 exports.handlers = {
   jsdocCommentFound: event => {
+    const templateMatch = /@template {(.*)} ([A-Z])\n/.exec(event.comment || '');
+
+    if (templateMatch) {
+      const className = templateMatch[1];
+      const templateName = templateMatch[2];
+
+      event.comment = event.comment.replace(new RegExp(`{(.*\\b)(${templateName})(\\b.*)}`, 'g'), `{$1typeof ${className}$3}`);
+    }
+
+    // \{.*\bT\b.*\}
```

Comment:
Do you think it's possible to modify the regex so that this case is also handled?

```javascript
    class CustomComponent extends Component{
      __z(){}
    }
    const Custom = Component.registerComponent('CustomComponent', CustomComponent);
    const custom = new Custom(player);
```

Instead of:
![Screenshot from 2025-02-16 19-57-07](https://github.com/user-attachments/assets/07df9b14-a181-4484-9c94-75bd406b3e44)

We would have:
![Screenshot from 2025-02-16 19-58-05](https://github.com/user-attachments/assets/9eac91b7-6aa5-47d8-9acb-10bbbd46872f)

Reply:
Thanks! The latest commit should cover that.